request.go

```diff
@@ -38,11 +39,26 @@ type MessageImprint struct {
 }
 
 // Equal compares if m and n are the same MessageImprint
+//
+// Reference: RFC 3161 2.4.2
 func (m MessageImprint) Equal(n MessageImprint) bool {
 	return m.HashAlgorithm.Algorithm.Equal(n.HashAlgorithm.Algorithm) &&
+		m.HashAlgorithm.Parameters.Class == n.HashAlgorithm.Parameters.Class &&
+		m.HashAlgorithm.Parameters.Tag == n.HashAlgorithm.Parameters.Tag &&
+		m.HashAlgorithm.Parameters.IsCompound == n.HashAlgorithm.Parameters.IsCompound &&
+		bytes.Equal(m.HashAlgorithm.Parameters.Bytes, n.HashAlgorithm.Parameters.Bytes) &&
+		bytes.Equal(m.HashAlgorithm.Parameters.FullBytes, n.HashAlgorithm.Parameters.FullBytes) &&
```

Comment:
Please refactor this block of code to some internal methods to compare two `asn1.RawValue`.

Reply:
Updated.